File: moderate/DETECTING-CYCLES/DETECTING CYCLES.java
import java.io.*;
public class Main {
    public static void main(String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            Cycle(line);
            // Process line of input Here
        }
    }

    private static void Cycle(String line){
        String result = "";
        String[] numsStr = line.split(" ");
        int leng = numsStr.length; 
        if(leng >= 2){
            if(leng == 2){
                if (numsStr[0].equals(numsStr[1])) result = numsStr[0];
            }else {
                int i = 0;
                boolean found = false;
                while(!found && i<leng-1){
                    int j = i+1;
                    while (!found && j<leng){
                        if (numsStr[i].equals(numsStr[j])) {
                            if (leng-j>=j-i){
                                result = getCycle(numsStr,i,j);
                                found = result.length() > 0;
                            }
                        }
                        j++;
                    }
                    i++;
                }
            }
        }
        System.out.println(result.trim());
    }
    
    private static String getCycle(String[] numsStr, int i, int j) {
        String result = "";
        int index = 0;
        int leng = numsStr.length;
        boolean isCycle = true;
        while (i+index<j && j+index<leng && isCycle){
            if(numsStr[i+index].equals(numsStr[j+index])){
                result += numsStr[i+index] + " ";
                index++;
            } else isCycle = false;
        }
        if(!isCycle || i+index!=j) result = ""; 
        return result;
    }
}

File: moderate/DetectingCycles/Detecting Cycles.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            Cycle(line);
        }
    }
    
    private static void Cycle(string line){
        string result = "";
        string[] numsStr = line.Split(' ');
        int leng = numsStr.Length; 
        if(leng >= 2){
            if(leng == 2){
                if (numsStr[0].Equals(numsStr[1])) result = numsStr[0];
            }else {
                int i = 0;
                bool found = false;
                while(!found && i<leng-1){
                    int j = i+1;
                    while (!found && j<leng){
                        if (numsStr[i].Equals(numsStr[j])) {
                            if (leng-j>=j-i){
                                result = getCycle(numsStr,i,j);
                                found = result.Length > 0;
                            }
                        }
                        j++;
                    }
                    i++;
                }
            }
        }
        Console.WriteLine(result.Trim());
    }
    
    private static string getCycle(string[] numsStr, int i, int j) {
        string result = "";
        int index = 0;
        int leng = numsStr.Length;
        bool isCycle = true;
        while (i+index<j && j+index<leng && isCycle){
            if(numsStr[i+index].Equals(numsStr[j+index])){
                result += numsStr[i+index] + " ";
                index++;
            } else isCycle = false;
        }
        if(!isCycle || i+index!=j) result = ""; 
        return result;
    }
}
